sqsjobs/item.go

```diff
@@ -161,36 +168,48 @@ func (i *Item) Nack() error {
 		return nil
 	}
 
-	// requeue message
-	err := i.Options.requeueFn(context.Background(), i)
-	if err != nil {
-		return err
-	}
+	if !i.Options.RetainFailedJobs {
+		// requeue as new message
+		err := i.Options.requeueFn(context.Background(), i)
+		if err != nil {
+			return err
+		}
+		// Delete original message
+		_, err = i.Options.client.DeleteMessage(context.Background(), &sqs.DeleteMessageInput{
+			QueueUrl:      i.Options.queue,
+			ReceiptHandle: i.Options.receiptHandler,
+		})
 
-	_, err = i.Options.client.DeleteMessage(context.Background(), &sqs.DeleteMessageInput{
-		QueueUrl:      i.Options.queue,
-		ReceiptHandle: i.Options.receiptHandler,
-	})
+		if err != nil {
+			return err
+		}
+	} else if i.Options.ErrorVisibilityTimeout > 0 {
+		// If error visibility is defined change the visibility timeout of the job that failed
+		_, err := i.Options.client.ChangeMessageVisibility(context.Background(), &sqs.ChangeMessageVisibilityInput{
+			QueueUrl:          i.Options.queue,
+			ReceiptHandle:     i.Options.receiptHandler,
+			VisibilityTimeout: i.Options.ErrorVisibilityTimeout,
+		})
 
-	if err != nil {
-		return err
-	}
+		if err != nil {
+			var notInFlight *types.MessageNotInflight
+			if goErrors.As(err, &notInFlight) {
+				// I would like to log info/warning here, but not sure how to get the correct logger
+				log.Println("MessageNotInFlight; ignoring ChangeMessageVisibility")
+			} else {
+				return err
```

Comment:
Please advise on how to correctly log this.

Reply:
Commented above